pkg/controller/tidbmonitor/tidb_monitor_control.go

```diff
@@ -57,7 +57,7 @@ func (c *defaultTidbMonitorControl) ReconcileTidbMonitor(tm *v1alpha1.TidbMonito
 func (c *defaultTidbMonitorControl) reconcileTidbMonitor(tm *v1alpha1.TidbMonitor) error {
 	var errs []error
 	oldStatus := tm.Status.DeepCopy()
-	if err := c.monitorManager.SyncMonitor(tm); err != nil {
+	if err := c.monitorManager.SyncMonitor(tm.DeepCopy()); err != nil {
```

Comment:
If deepcopy here, the status in L64 will be the old status, just deepcopy here https://github.com/pingcap/tidb-operator/blob/master/pkg/controller/tidbmonitor/tidb_monitor_controller.go#L117.

Reply:
I think we should get old status to compare new status.